CCPA/Data Deletion Request Handling.md

```diff
@@ -0,0 +1,329 @@
+*   The iframe is hosted on the publisher domain
+*   If the iframe sandbox parameter is provided, the following restrictions are lifted:
+    *   `allow-scripts`
+    *   `allow-storage-access-by-user-activation`
+
+
+## registerDeletion
+
+This command registers a vendor-specific callback function at the API. When the publisher loads the vendor deletion script, those scripts must execute `registerDeletion`. 
+
+This command is an update to the existing `__uspapi` function with the following syntax: 
+
+```
+__usapi("registerDeletion", version, performDeletionFunction)
+```
+
+Upon execution, the vendor script is registered to receive the signal sent by the call to `performDeletion`.
+
+
+## performDeletion
+
+The publisher, or its CMP where applicable, calls this command based on some user delete action to initiate the deletion process. The command communicates to vendors that a specified user has requested that their personal data be deleted by calling the `performDeletionFunction` registered during the call to `registerDeletion`. 
+
+This command is an update to the existing `__uspapi` function with the following syntax: 
+
+```
+__usapi("performDeletion", version, null, identifiers)
+```
+
+The callback parameter of the `__uspapi` is not used in this case and can remain null. The `identifiers` parameter is only required when handling non-web deletion requests and is further explained in the following section. 
+
+
+### Identifying Data to Delete in Non-web Environments
+
+When operating in a non-web environment, data deletion requests are handled by sending the user to a web page where they can complete the request to have their data deleted. Vendors need certain information to correctly identify the data to delete:  the platform name, the unique app identifier used in the app store, and the device identifier for that platform / store. 
+
+The `performDeletion` command includes a parameter for identifiers. Using the `identifiers` parameter, publishers can pass multiple items, each with the required fields: `platform`, `app_identifier`, and `user_identifier`. These details should be passed from the app context to the delete webpage and then along with the request from the page. Below are a few examples of that information:
+
+```
+identifiers {
+  ...
+  "platform": "ios-app-store",
+  "app_identifier": "01234567891446075923",
+  "user_identifier": "AEA12347583AACD-A123667-A418AABC-AB123806-1242AEAACB12AB1234548606"
+}
+...
+{
+  "platform": "android-play-store",
+  "app_identifier": "com.acmeinc.acmeapp",
+  "user_identifier": "aaaacdda123802ae-11fb191c-1247abad-12340794ad123394ac123912"
+...
+}
+```
+
+```
+Note: For purposes of this explanation, a "bundle id" is used for (iOS) and a "package" for (Android).
+```
+
+### Common Platform / Stores identifiers
+
+These identifiers can be used for the "platform" field where applicable. Otherwise, the platform field can include an arbitrary value.
```

Comment:
How can `platform` contain an arbitrary value?  If a Vendor is meant to interpret this information to be able to identify a user, this can not be arbitrary.

Reply:
We've decided to keep it open in order to support additional platforms that aren't listed (knowing that we can't possibly list them all)

Reply:
@Facens if it's arbitrary it's not usable because this information enables a Vendor script to interpret the other two values.  For example: If I'm a Vendor and I receive a `"platform"` value of `"foo-bar-23"` and it's not a platform I recognize, what do I do?